_posts/2020-11-09-Basics-With-Bioawk.md

```diff
@@ -0,0 +1,62 @@
+
+Awk programs are, generally, built using the following recipe:
+
+`pattern {action}`
+
+The _pattern_ is an expression or regular expression patter, allowing people to filter for matched expressions, so that those only those patterns are acted upon. These _pattern-actions_ can be linked together into a chain by semicolons. By combining these two different concepts, _records_ and _fields_ with _pattern-actions_, Awk is able survey a file line by line and split those lines into fields, compare those lines and fields to a pattern, and perform actions on those patters to manipulate data and generate reports.
+
+Some common Awk functions are to filter reports and fields using RegEx, splitting strings (s) into tabular "chunks" with a specificed delimiter (d) and placing them into an array (x) via `split (s, x, d)`, `print` the content of the tabular data, changing the case of strings in a field via `tolower(s)` or `toupper(s)`
+
+Now that we've lightly sautéed the awk, so to speak, let's add the "bio" back to "Bioawk."
+
+## Bioawk: Making Biological Formats More Awk Since 2015
+
+Like Awk, Bioawk processes data, but extends Awk's "flavor compounds" to common biological file formats, particularly FASTA/FASTQ, GTF/GFF, BED, SAM, and VCF file types. The beauty of Bioawk is that it will automatically set the variables for each field to particular parts of the bioinformatics files. Some examples of this are as follows:
+```
+bed:
+
+    1:chrome 2:start 3:end 4:name 5:score 6:strand 7:thickstart 8:thickend
+    9:rgb 10:blockcount 11:blocksizes 12:blockstarts`
+
+gff:
+
+    1:seqname 2:source 3:feature 4:start 5:end 6:score 7:filter 8:strand
+    9:group 10:attribute
+
+fastx:
+    1:name 2:seq 3:qual 4:comment
+```
+Bioawk code ~tastes~ is built similarly to Awk, but by using bioawk's built in tools, we can specify the filetype we are using with the following syntax, using FASTX as an example: `bioawk -c fastx`
+
+We will continue to FASTA/FASTQ files as an example moving forward. As mentioned above, Bioawk has built in tooling to scan a file and "chunk" parts of the sequence file into their appropriate fields, the name of the sequence (`$name`), the sequence itself `($seq)`, comments about the sequence such as what the gene does, what chromosome it is on, etc. (`$comment`), and finally FASTQ files have an additional field for quality data about the sequence (`qual`). If we wanted to print the comments of all the sequences in a FASTA file, we could write the following:
```

Comment:
I think here you might have omitted a word as well. You write "we will continue to FAST/FASTQ files ...", do you mean to write "we will continue to use FAST/FASTQ files ..." ?